frontend/assignments/2/frontend/script/htmlScript/code.js

```diff
@@ -0,0 +1,587 @@
+      console.log("input length");
+      postButton.style.backgroundColor = "#1D9BF0";
+      postButton.style.color = "white";
+    } else {
+      postButton.style.backgroundColor = "";
+      postButton.style.color = "";
+    }
+  });
+
+  floatingTweetBoxIcon.addEventListener("click", function () {
+    if (tweetBoxProfileMobile.style.display === "none") {
+      console.log("here");
+      tweetBoxProfileMobile.style.display = "block";
+      posts.style.display = "none";
+      tweetBox.style.display = "flex";
+
+      tweetInput.style.display = "block";
+      profileIcon.style.display = "none";
+      floatingTweetBoxIcon.style.display = "none";
+    } else {
+      tweetBoxProfileMobile.style.display = "none";
+    }
+  });
+
+  let backArrow = document.querySelector(".back-arrow");
+  backArrow.addEventListener("click", function () {
+    tweetBoxProfileMobile.style.display = "none";
+    posts.style.display = "block";
+    tweetInput.style.display = "block";
+    profileIcon.style.display = "flex";
+    floatingTweetBoxIcon.style.display = "flex";
+    tweetBox.style.display = "none";
+  });
+
+  postButton.addEventListener("click", function () {
+    const tweetContent = tweetInput.value.trim();
+    console.log("clicked: ", tweetContent);
+    if (tweetContent.length > 0) {
+      let img = null;
+
+      if (isImageUploaded()) {
+        img = document.querySelector(".post-input-image img").src;
+      }
+      console.log(imagePath)
+      const newPost = {
+        username: username,
+        timestamp: new Date().getTime(),
+        message: tweetContent,
+        img: img == null ? null : img,
+      };
+
+      const newPostJSON = JSON.stringify(newPost);
+
+      // Calculate the payload size in bytes
+      const payloadSizeInBytes = new Blob([newPostJSON]).size;
+    
+      console.log("Payload size of newPost object:", payloadSizeInBytes, "bytes");
+      console.log(newPost);
+      fetch(`http://${window.location.hostname}:3000/api/posts`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(newPost),
+      })
+        .then((response) => {
+          if (response.ok) {
+            return response.json();
+          } else {
+            throw new Error("Failed to add post");
+          }
+        })
+        .then((post) => {
+          createPostElement(post);
+          if (window.innerWidth < 500) {
+            tweetBoxProfileMobile.style.display = "none";
+            posts.style.display = "block";
+            profileIcon.style.display = "flex";
+            floatingTweetBoxIcon.style.display = "flex";
+            tweetBox.style.display = "none";
+          }
+          window.location.reload();
+
+          tweetInput.value = "";
+          postButton.style.backgroundColor = "";
+          postButton.style.color = "";
+        });
+    }
+  });
+
+ 
+  document.addEventListener("click", function (event) {
```

Comment:
Rather than doing this lets have a single click listener and use switch for covering all cases.